src/main/java/io/leonis/subra/gui/field/PotentialBackground.java

```diff
@@ -0,0 +1,25 @@
+package io.leonis.subra.gui.field;
+
+import com.googlecode.lanterna.*;
+import java.util.function.Function;
+import org.nd4j.linalg.api.ndarray.INDArray;
+
+/**
+ * @author Jeroen de Jong
+ */
+public class PotentialBackground implements Function<TerminalPosition, TextColor> {
+  private final Function<Double, TextColor.RGB> gradient;
+  private final INDArray data;
```

Comment:
i think the sampling mechanism (`FieldGenerator`) should be part of this class or be generalized and move to `algieba`

Reply:
The idea is that this `data` field is acquired at the end of our potentialfield-strategy. I don't know how we can effectively couple it to be honest.

Reply:
i'm not sure what you mean by that, because the value is calculated in single points if it's needed for the strategy. if you need a sampling of the entire field then that is unrelated to the strategy

Reply:
Yeah but essentially we need to calculate a a subset of all values for the gui; so the gui should receive or be able to retrieve the necessary dataset at one point.